<!--
 Copyright 2022 Shadow Robot Company Ltd.
 This program is free software: you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the Free
 Software Foundation version 2 of the License.
 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 more details.
 You should have received a copy of the GNU General Public License along
 with this program. If not, see <http://www.gnu.org/licenses/>.
-->

<launch>
    <arg name="rviz" default="false"/>
    <arg name="right_root_tf_name" default="sr_leap_motion_root"/>
    <arg name="left_root_tf_name" default="sr_leap_motion_root"/>
    <arg name="bone_starts" default="false"/>
    <arg name="bone_middles" default="false"/>
    <arg name="heartbeat_topic" default="/sr_leap_motion/heartbeat"/>
    <include file="$(find sr_leap_motion)/launch/leap_motion.launch"/>
    <node pkg="sr_leap_motion" type="sr_leap_motion.py" name="sr_leap_motion" output="screen">
        <param name="right_root_tf_name" value="$(arg right_root_tf_name)"/>
        <param name="left_root_tf_name" value="$(arg left_root_tf_name)"/>
        <param name="bone_starts" value="$(arg bone_starts)"/>
        <param name="bone_middles" value="$(arg bone_middles)"/>
        <param name="heartbeat_topic" value="$(arg heartbeat_topic)"/>
    </node>
    <!-- Open a preconfigured RViz window -->
    <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(find sr_leap_motion)/config/rviz.rviz" />
</launch>
